eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisher.scala

```diff
@@ -239,11 +239,11 @@ private class ReplaceableTxPublisher(nodeParams: NodeParams,
   }
 
   def checkHtlcPreconditions(replyTo: ActorRef[TxPublisher.PublishTxResult], cmd: TxPublisher.PublishReplaceableTx, htlcTx: HtlcTx, targetFeerate: FeeratePerKw): Behavior[Command] = {
-    val commitFeerate = cmd.commitments.localCommit.spec.feeratePerKw
+    val htlcFeerate = cmd.commitments.localCommit.spec.htlcTxFeerate(cmd.commitments.channelType)
```

Comment:
So, this is where the magic is happening right? But I don't get how it works. Why don't we go to the `fund()` branch when `targetFeerate != htlcFeerate`?

Reply:
I don't understand your question. What we're doing here is simply checking the current feerate of the htlc tx as it was signed by both parties during channel operation. If that feerate is >= targetFeerate, we don't need to add inputs/outputs and use the simpler flow of just signing the htlc tx and broadcasting it.

With `anchor_output_zero_fee_htlc_tx` the `htlcFeerate` here will always be `0`, so we'll always go to the `fund()` branch, but for anchor outputs without `zero_fee_htlc_tx` we may directly broadcast the htlc tx without any funding (which is better for our utxo set management).

Reply:
Nevermind 🙃